TodoList/Src/TodoList.Dependency/Extensions/LifetimeExtension.cs

```
@@ -0,0 +1,23 @@
+﻿using TodoList.Contracts.DI;
+using Unity.Lifetime;
+
+namespace TodoList.DI.Extensions
+{
+    internal static class LifetimeExtension
+    {
+        internal static LifetimeManager GetUnityLifetimeManager(this Lifetime lifetime)
+        {
+            switch (lifetime)
+            {
+                case Lifetime.PerApplication:
+                    return new ContainerControlledLifetimeManager();
+
+                case Lifetime.PerRequest:
+                    return new HierarchicalLifetimeManager();
+
+                default:
+                    return new TransientLifetimeManager(); // default in UnityContainer.RegisterType()
```

From the `extension-on-Lifetime` perspective, this is an unexpected value. You should not "correct" the lack of functionality by guessing the least harmful result – [fail fast](https://dzone.com/articles/fail-fast-principle-in-software-development). Throwing an `InvalidOperationException` seems much more reasonable in this situation.

This `default` gets typically hit when you extend the extended type in question (`Lifetime` in this case) in future but you forget/don't know about the extension method. Imagine that somebody extended the type with `.PerMonth` (for what-ever ill reason). In such a case, it would be unfortunate to create instance "`.PerInvocation`", wouldn't it? 😉